runtime/src/main/java/org/corfudb/runtime/view/CorfuGuidGenerator.java

```diff
@@ -0,0 +1,43 @@
+package org.corfudb.runtime.view;
+
+import org.corfudb.protocols.wireprotocol.Token;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Ordered Globally Unique Identity (GUID) generator that returns ids
+ * having a notion of a universal comparable ordering with the help of
+ * a counter inside the Corfu SequencerServer.
+ * Should the Sequencer restart, it will come up with a higher epoch number.
+ * Load this epoch into the higher bits of the guid and load the counter
+ * into the lower bits to form the globally unique ordered id.
+ *
+ * Created by Sundar Sridharan on 5/22/19.
+ */
+public class CorfuGuidGenerator implements OrderedGuidGenerator {
+    // 3 bytes for epoch ~= 16.7 million epoch changes before rollover.
+    private final static int MAX_EPOCH = 0xFFffFF;
+    private final static int SHIFT_EPOCH = 40;
+    // 5 bytes for sequence ~= 1 trillion sequence changes per epoch before rollover.
+    private final static long MAX_SEQUENCE = 0xFFffFFffFFL;
+    CorfuRuntime rt;
+    public CorfuGuidGenerator(CorfuRuntime rt) {
+        this.rt = rt;
+    }
+
+    @Override
+    public long nextLong() {
+        TokenResponse token = rt.getSequencerView().nextOrderedGuid();
```

Comment:
this looks strange. Can you please explain why  `nextOrderedGuid ` situated inside SeqencerView?

Reply:
it is a wrapper around the SequencerView's nextOrderedGuid()